packages/flutter_tools/lib/src/android/migrations/gradle_java_version_conflict_migration.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../base/version.dart';
+import '../../project.dart';
+import '../android_studio.dart';
+
+// Android Studio 2022.2 "Flamingo" is the first to bundle a Java 17 JDK.
+// Previous versions bundled a Java 11 JDK.
+final Version _androidStudioFlamingo = Version(2022, 2, 0);
+final RegExp _distributionUrlMatch = RegExp(r'^\s*distributionUrl\s*=');
+final RegExp _gradleVersionMatch = RegExp(
+  r'^\s*distributionUrl\s*=\s*https\\://services\.gradle\.org/distributions/gradle-((?:\d|\.)+)-(?:all|bin)\.zip');
```

Comment:
I changed it to share the parse logic in the linked pr, but am now thinking about changing it back to how it was before, because: 

if we only make changes to gradle.org urls, we get a guarantee that the new url corresponds to an actual hosted zip (because gradle.org hosts all versions of gradle). 

But I am not sure if that is safe to assume with other distribution urls, e.g. someone who had for some reason set up a personal distribution url, or a company that isn't regularly updating.

Reply:
> if we only make changes to gradle.org urls, we get a guarantee that the new url corresponds to an actual hosted zip (because gradle.org hosts all versions of gradle).
> 
> But I am not sure if that is safe to assume with other distribution urls, e.g. someone who had for some reason set up a personal distribution url, or a company that isn't regularly updating.

Hmm, yeah, that reasoning makes a lot of sense.

I think using a distribution URL that isn't on gradle.org is likely to be pretty uncommon, too. So that'd mean there's not much cost to playing it safe with those and not trying to update them.